angular2-jwt.ts

```diff
@@ -56,57 +64,74 @@ export class AuthHttp {
 
   private _config: IAuthConfig;
   public tokenStream: Observable<string>;
+  private jwtHelper:JwtHelper;
 
   constructor(options: AuthConfig, private http: Http) {
     this._config = options.getConfig();
+    this.jwtHelper = new JwtHelper();
 
     this.tokenStream = new Observable((obs: any) => {
       obs.next(this._config.tokenGetter())
     });
   }
 
-  _request(url: string | Request, options?: RequestOptionsArgs) : Observable<Response> {
+  _request(req: Request) : Observable<Response> {
 
     let request:any;
-    
-    if(!tokenNotExpired(null, this._config.tokenGetter())) {
-      if(!this._config.noJwtError) {
-        throw 'Invalid JWT';
+
+    // this is observable as it may need to come from an http call ...
+    let refreshToken:Observable<string>;
+
+    var token = this._config.tokenGetter()
+    // do we even have a token?
```

Comment:
Your comments mostly say what the next line is doing, not why it is doing it. Essentially your comment repeats the code, forcing multiple lines to be maintained for each actual line of code.